psychopy/hardware/pr.py

```diff
@@ -356,55 +365,24 @@ def __del__(self):
     def startRemoteMode(self):
         """Sets the Colorimeter into remote mode
         """
-        reply = self.sendMessage(b'PHOTO', timeout=10.0)
+        reply = self.sendMessage('PHOTO', timeout=10.0)
 
     def getDeviceType(self):
         """Return the device type (e.g. 'PR-655' or 'PR-670')
         """
-        reply = self.sendMessage(b'D111')  # returns errCode,
+        reply = self.sendMessage('D111')  # returns errCode,
         return _stripLineEnds(reply.split(',')[-1])  # last element
 
     def getDeviceSN(self):
         """Return the device serial number
         """
-        reply = self.sendMessage(b'D110')  # returns errCode,
+        reply = self.sendMessage('D110')  # returns errCode,
         return _stripLineEnds(reply.split(',')[-1])  # last element
 
-    def sendMessage(self, message, timeout=0.5, DEBUG=False):
-        """Send a command to the photometer and wait an allotted
-        timeout for a response (Timeout should be long for low
-        light measurements)
-        """
-        # send complete message
-        msg = "Sending command '%s' to %s"
-        logging.debug(msg % (message, self.portString))
-        if message[-1] != '\n':
-            message += '\n'  # append a newline if necess
-
-        # flush the read buffer first
-        # read as many chars as are in the buffer
-        self.com.read(self.com.inWaiting())
-
-        # send the message
-        for letter in message:
-            # for PR655 have to send individual chars ! :-/
```

Comment:
Should this have been removed? It looks like it causes [this](https://discourse.psychopy.org/t/cannot-connect-photoresearch-pr655-to-pc-using-psychopy-communication-protocol-problem/25595/12) issue.

Reply:
Tagging @peircej

Reply:
This is reported as being fixed, here: https://github.com/psychopy/psychopy/issues/5237 / https://github.com/psychopy/psychopy-photoresearch/commit/872718f3198f317c3dc3c9dc939b8c2f6d1bec32